### DISCUSSION TITLE
AFJ new name

### DISCUSSION THREAD

**Discussion by genaris** [openwallet-foundation/credo-ts#1668] [Polls]
Following the transition to OWF, we need to find a good name for AFJ ASAP. 

After some time digging around to check for good candidates that are not taken by any other project (at least not in our industry), we came up with the two finalists. Each of them has its own pros / cons that I'll summarize below. Please let us know your preference before our next WG meeting (January 4th 2024)!

**Credo**
Latin word for “I believe” or “I trust”. Also a nod to the word “Credential”. 

Possible npm package names: @credo/core, @credo/didcomm, @credo/openid4vc-client, etc.

Pros: 
- It is cool!
- It's trademarkable

Cons:
- It is very much related to religion in some languages e.g. Italian, Spanish, French, Portuguese and is more used in the context of faith/religion
- NPM scope @credo is already [taken](https://www.npmjs.com/~credo) by [Credo360](https://github.com/herculesinc/credo.faq/blob/master/content/general.md), although it seems unmaintained since 5 years now and might be possible to take it over
	- alternative could be `@credo-id`

**Open Trust Framework** 
(or just Open Trust) Self-explanatory name of this project.

Possible npm package names: @open-trust/core, @open-trust/didcomm, @opentrust/oidc4vc, etc.

Pros:
- Simple name
- After a quick search, despite its simplicity there is no other related project called like that
- NPM scopes @open-trust and @opentrust are available

Cons:
- Not trademarkable
- Verbose in NPM package names

**Comment by kukgini** [comment]
It would be nice to be considered along with the name of this project: https://github.com/openwallet-foundation-labs/wallet-framework-dotnet

**Reply by TimoGlastra** [reply]
Yes that's a good suggestion. One thing to consider is the dotnet framework is mostly about wallets, while this one is also about server side deployments.

**Reply by kukgini** [reply]
I don't know if the current .NET has changed its goals. However, originally, .NET was also a framework that enabled server side deployment.

**Comment by tkuhrt** [comment]
You may want to consider a name that can be trademarked. Of course, this will mean working with LF Legal to determine if the project name chosen can be trademarked. Currently, all of the names above will most likely not be able to be trademarked.

**Comment by tkuhrt** [comment]
Possible options:
* Fides - goddess represents also a fundamental value for the romans, that of loyalty and trust between two parts
* Hamal - the brightest star in the constellation Aries (a nod to your roots)

**Reply by genaris** [reply]
I really love Fides, it is a perfect match for us! But unfortunately it is already taken by another project (https://github.com/ethyca/fides).

**Reply by tkuhrt** [reply]
I was only checking Google. Searched for the name + wallet to see if there were any wallet projects named that. Apologies for getting your hopes up.

**Reply by kukgini** [reply]
Bifold - from Aries Framework Javascript and React Native.

**Reply by karimStekelenburg** [reply]
How about Fiducia? If I'm not mistaking Fides is mostly used to in the context of faith, where Fiducia puts an emphasis on confidence.

**Reply by genaris** [reply]
Fiducia sounds good to me!

**Reply by berendsliedrecht** [reply]
Not too opposed to fiducia, but also not the biggest fan. I will also have a think about it as well!

**Comment by kukgini** [comment]
openwallet-framework-js
openwallet-framework-dotnet

**Comment by sparkandflash** [comment]
aeorium-wallet

**Comment by swcurran** [comment]
Jazz Agent

**Comment by tkuhrt** [comment]
confisus - Latin for confident

**Comment by TimoGlastra** [comment]
Suggestion from my colleague Inge: Credo

**Reply by TimoGlastra** [reply]
My favourite option at the moment

**Reply by berendsliedrecht** [reply]
Yes! Credo is my favourite as well

**Reply by berendsliedrecht** [reply]
I did just see that the namespace of credo is taken on npm sadly.

**Comment by genaris** [comment]
I've just updated the poll with the two finalists:

- Credo
- Open Trust

**Comment by jorgefl0** [comment]
Late to the party but here is my proposal: HAL (Human Agent Libraries) 😃

**Comment by TimoGlastra** [comment]
Discussed on the Credo WG call of 18-01-2024. 

We're going with Credo as it has the most votes. Open Trust Framework is a good additional description, tagline:
"Credo: The Open Trust Framework"

Scope will be `@credo-id`.

**Reply by berendsliedrecht** [reply]
How about the scope `@credojs`?

**Reply by TimoGlastra** [reply]
To me it feels like the name then reads as credojs (kredois). But not opposed to it

I think it might also be nice to rename core to credo. So it becomes `@credojs/credo` for the main package. And core could be a lower level package where shared utils are placed in. 

So one of: `@credo-js`, `@credojs`, `@credo-id`, `@credo.js`

**Reply by TimoGlastra** [reply]
Registred `credojs` and `credo-id` for now. It seems `credo-js` is already registered ?!?

**Reply by berendsliedrecht** [reply]
Does `@credo.js` work? Can we use a period? That would be the best looking, but maybe a bit more annoying to type.

**Reply by TimoGlastra** [reply]
Yes that works. Just registered it :)

**Comment by ericvergnaud** [comment]
Credo is a good name! FYI I'm not sure credo is still associated with religion in France. It definitely originates from christianism, but it's now commonly understood as "Principles on which we base our opinion and our conduct." I suspect the Wikipedia entry is a bit misleading.

**Comment by TimoGlastra** [comment]
Done :)
